controller/pagecontroller.php

```diff
@@ -95,8 +95,8 @@ function __construct(
 	 * @NoCSRFRequired
 	 */
 	function index(string $userFilter = "") {
-		if(\OC::$server->getGroupManager()->isInGroup($this->userId, "provider"))
-			return;
+		if(\OC::$server->getGroupManager()->isInGroup($this->userId, "provider"))		
+			return new RedirectResponse('/apps/adminly_dashboard');
```

Comment:
It would be slightly better if instead of redirectly directly to the Dashboard, we redirect to the base url, since that will automatically go to the default app if the user is logged in.

The URLgenerator is retrieved in this function, so you can move that one up to the top, and then use that to retrieve the baseURL for the instance.

Reply:
I've set the urlGenerator variable as global to make the code cleaner

Reply:
Nice!